### ISSUE TITLE
process.cpu_times() different across versions.

### ISSUE THREAD

**Issue by rezmont** [giampaolo/psutil#1110]
I have two versions of the psutil on the same box and they seem to be reporting different CPU times, for the same pid.

One is version 2.2.1 and the other one is 5.2.0.

Interestingly the one running 5.2.0 cpu_times sometimes decreases. I was under the impression that cpu_times are only timers that should be always increasing.

**Comment by giampaolo** [comment]
What platform are you on? What's the value which decreases?

**Comment by rezmont** [comment]
centos7

**Comment by giampaolo** [comment]
What's the value which decreases?

**Comment by rezmont** [comment]
t1:20:37:10.152525 ProcessSample(rate=0.0, user=521.55, system=952.61, rss=35430400, vms=1600204800, nice=0, status='sleeping', threads=20)
t2:20:42:58.449567 ProcessSample(rate=0.0, user=322.65, system=577.12, rss=35430400, vms=1600204800, nice=0, status='sleeping', threads=20)

**Comment by rezmont** [comment]
What I pasted above is from my log. That code uses 5.2.0.

**Comment by rezmont** [comment]
I am also using the oneshot context.

**Comment by giampaolo** [comment]
Are you able to reproduce this in a reliable fashion? You should `cat /proc/{pid}/stat`, check the 10th and 11st value and see if the numbers decreases. If they do then it's a kernel bug.

**Comment by rezmont** [comment]
Let me put a watch on that and get back to you in a bit. Any idea why the older version is reporting totally different numbers:
pcputimes(user=525.49, system=959.46)

**Comment by rezmont** [comment]
I think you mean 12 and 13? at least based on what it says here?

**Comment by giampaolo** [comment]
Yes, sorry, 14 and 15.

**Comment by rezmont** [comment]
```
~ $ date; cat /proc/866/stat | awk '{print $14, $15}'
Wed Jul 19 21:55:46 UTC 2017
52691 96172
```

The values seem to agree with one of the samples. Also this is totally reproducible. I just need to start periodically monitoring all pids on my server.

**Comment by giampaolo** [comment]
Just bear in mind that psutil divides the values by clock ticks (`100` on Ubuntu 16.04).

**Comment by rezmont** [comment]
I understand that. It does not explain why it reads different values tho. I just started tailing my logs and as samples came in catted the proc/<pid>/stat.


```
D0719 22:01:37.055901 24474 process_collector_psutil.py:105] 866: ProcessSample(rate=0.0, user=332.69, system=588.16, rss=35430400, vms=1600204800, nice=0, status='sleeping', threads=20)
D0719 22:06:22.156851 24474 process_collector_psutil.py:105] 866: ProcessSample(rate=0.0, user=329.27, system=590.79, rss=34787328, vms=1600204800, nice=0, status='sleeping', threads=20)
```

```
~ $ date; cat /proc/866/stat | awk '{print $14, $15}'
Wed Jul 19 22:01:41 UTC 2017
52733 96241
~ $ date; cat /proc/866/stat | awk '{print $14, $15}'
Wed Jul 19 22:06:27 UTC 2017
52766 96296
```

**Comment by rezmont** [comment]
I was wondering if there a known issue when `oneshot` is used in multiple threads repeatedly?

**Comment by giampaolo** [comment]
So, oneshot is *not* thread-safe at the moment, but I don't think it has to do with this issue. AFAIU the worst that can happen is that another thread may "exit" the ctx manager sooner than expected, and as such the info retrieval will be slower but that is it: the returned data is supposed to remain correct.

**Comment by giampaolo** [comment]
I've added a unit test which tests the parsing of /stat file here https://github.com/giampaolo/psutil/commit/28e0aef2d2a40b61dad7081ddaa88917abba52e7 and it seems everything is ok so I'm not sure what's wrong here. Maybe it's centos 7?

**Comment by rezmont** [comment]
I don't think it is parsing the file that causes the issue. I think it is the shared context across multiple processes/threads that are reading different pid's stats. Does this make sense?

**Comment by giampaolo** [comment]
I don't see how that can cause this issue. Please show me your code (I should have asked you earlier).
